mitosheet/src/mito/components/endo/celleditor/CellEditorDropdown.tsx

```diff
@@ -117,9 +117,11 @@ const CellEditorDropdown = (props: {
                                     return undefined
                                 }
                                 const prevEditingMode = {...prevEditorState}.editingMode
+                                const newEditingMode = prevEditingMode === 'entire_column' ? 'specific_index_labels' : 'entire_column'
+                                localStorage.setItem("editingMode",newEditingMode)
                                 return {
                                     ...prevEditorState,
-                                    editingMode: prevEditingMode === 'entire_column' ? 'specific_index_labels' : 'entire_column'
+                                    editingMode: newEditingMode
```

Comment:
I think we should avoid fragmenting _where_ state is stored. Currently, we store the editing state in `Mito.tsx` in the `editorState` -- and I think it would be a confusing call to duplicate some of this state. We'd have to be reasoning about _what_ lives _where_ - rather than just being able to find a single source of truth. 

Here's an idea:

Take this line of code:
```
    const [editorState, setEditorState] = useState<EditorState | undefined>(undefined);
```
In Mito.tsx and change it's type to `EditorState | ClosedEditorState`. Define `ClosedEditorState` next to `EditorState` in `types.tsx` -- and pretty much it should just store that it is closed, and also what the current editing mode is!

Then throughout the codebase, change the closed checks from !== undefined to `editorState.type !== 'closed'` -- and then you can use this old state when toggling to the new one.  

Let me know if you have questions about this!

Reply:
done